src/classes/USER_FrozenInActiveUser_TDTM_TEST.cls

```diff
@@ -0,0 +1,159 @@
+		userFrozenInActiveUserTDTM.run(null, null, TDTM_Runnable.Action.BeforeInsert, null);
+	}
+
+
+	/*********************************************************************************************************
+    * @description Tests the handling of the TDTM_Runnable.Action for BeforeUpdate
+    * @return void
+    **********************************************************************************************************/
+    private static testMethod void testRunOverrideBeforeUpdate() {
+		USER_FrozenInActiveUser_TDTM userFrozenInActiveUserTDTM = new USER_FrozenInActiveUser_TDTM();
+		userFrozenInActiveUserTDTM.run(null, null, TDTM_Runnable.Action.BeforeUpdate, null);
+	}
+
+
+	/*********************************************************************************************************
+    * @description Tests the handling of the TDTM_Runnable.Action for BeforeDelete
+    * @return void
+    **********************************************************************************************************/
+    private static testMethod void testRunOverrideBeforeDelete() {
+		USER_FrozenInActiveUser_TDTM userFrozenInActiveUserTDTM = new USER_FrozenInActiveUser_TDTM();
+		userFrozenInActiveUserTDTM.run(null, null, TDTM_Runnable.Action.BeforeDelete, null);
+	}
+
+
+	/*********************************************************************************************************
+    * @description Tests the handling of the TDTM_Runnable.Action for AfterInsert 
+    * @return void
+    **********************************************************************************************************/
+    private static testMethod void testRunOverrideAfterInsert() {
+		USER_FrozenInActiveUser_TDTM userFrozenInActiveUserTDTM = new USER_FrozenInActiveUser_TDTM();
+		userFrozenInActiveUserTDTM.run(null, null, TDTM_Runnable.Action.AfterInsert, null);
+	}
+
+
+	/*********************************************************************************************************
+    * @description Tests the handling of the TDTM_Runnable.Action for AfterUpdate 
+    * @return void
+    **********************************************************************************************************/
+    private static testMethod void testRunOverrideAfterUpdate() {
+		USER_FrozenInActiveUser_TDTM userFrozenInActiveUserTDTM = new USER_FrozenInActiveUser_TDTM();
+
+		List<User> newUsers = createUsers();
+
+		// Freeze a User
+        List<UserLogin> currentUserLogin = [SELECT IsFrozen, UserId FROM UserLogin WHERE UserId = :newUsers[1].Id];
+		currentUserLogin[0].IsFrozen = true;
+		Database.update(currentUserLogin, false);
+
+		Test.startTest();
+		userFrozenInActiveUserTDTM.run(newUsers, null, TDTM_Runnable.Action.AfterUpdate, null);
+		Test.stopTest();
+
+		Error_Settings__c errorSettingsSysAdmin = UTIL_CustomSettingsFacade.getErrorSettings();
+        System.assertEquals(ERROR_NOTIFICATION_RECIPIENT_ALL_SYS_ADMINS, errorSettingsSysAdmin.Error_Notifications_To__c);
```

Comment:
In general, we've tested our TDTM logic by always doing the actual DML that should fire the TDTM logic.  by invoking the run method directly, you are not 100% verifying that a given DML operation has the intended effect that we expect.